datafusion/physical-optimizer/src/enforce_sorting/sort_pushdown.rs

```diff
@@ -114,6 +118,18 @@ fn pushdown_sorts_helper(
             sort_push_down.data.fetch = fetch;
             sort_push_down.data.ordering_requirement =
                 Some(OrderingRequirements::from(sort_ordering));
+            let filter = plan
+                .as_any()
+                .downcast_ref::<SortExec>()
+                .and_then(|s| s.filter().clone());
```

Comment:
This is... not the prettiest. And is a smell that it will be hard for 3rd party plans to implement this.

Reply:
I think all we need to do is add a method to the `ExecutionPlan` like `fn dynamic_filters` and it will be general and well documented